app/src/store/features/environmentVariables/slice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { EnvironmentVariableValue } from "backend/environmentVariables/types";
+import { ReducerKeys } from "store/constants";
+import getReducerWithLocalStorageSync from "store/getReducerWithLocalStorageSync";
+
+const initialState = {
+  variables: {} as Record<string, Record<string, EnvironmentVariableValue>>,
+};
+
+const slice = createSlice({
+  name: ReducerKeys.ENVIRONMENT_VARIABLES,
+  initialState,
+  reducers: {
+    resetState: () => initialState,
+    setVariables: (
+      state,
+      action: PayloadAction<{
+        newVariables: Record<string, EnvironmentVariableValue>;
```

Comment:
Make `Record<string, EnvironmentVariableValue>` as a type `EnvVar` so that we can use it everywhere for consistency